subiquity/tests/api/test_api.py

```diff
@@ -0,0 +1,521 @@
+#!/usr/bin/env python3
+
+import aiohttp
+from aiohttp.client_exceptions import ClientResponseError
+import async_timeout
+import asyncio
+from functools import wraps
+import json
+import logging
+import os
+import sys
+import unittest
+
+from subiquitycore.utils import astart_command
+
+
+logging.basicConfig(level=logging.DEBUG)
+socket_path = '.subiquity/socket'
+
+
+ver = sys.version_info
+if ver.major < 3 or ver.minor < 8:
+    raise Exception('skip asyncio testing')
+
+
+def find(items, key, value):
+    for item in items:
+        if item[key] == value:
+            yield item
+
+
+def first(items, key, value):
+    return next(find(items, key, value))
+
+
+def timeout(_timeout):
+    def wrapper(coro):
+        @wraps(coro)
+        async def run(*args, **kwargs):
+            with async_timeout.timeout(_timeout):
+                return await coro(*args, **kwargs)
+        return run
+    return wrapper
+
+
+def json_print(json_data):
+    print(json.dumps(json_data, indent=4))
+
+
+def loads(data):
+    return json.loads(data) if data else None
```

Comment:
Can this check for data == "" rather than just "if data"? If that's the intent. I also kind of dislike the ternary operator form you use here but maybe I just need to get over that.